release-calendar/src/server/repository-service.ts

```diff
@@ -57,4 +57,58 @@ export class RepositoryService {
 
     return releaseQuery;
   }
+
+  include(arr: Release[], rel: Release): boolean {
+    for (let i = 0; i < arr.length; i++) {
+      if (arr[i].name === rel.name) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  async getCurrentReleases(): Promise<Release[]> {
+    const rollbackQuery = this.releaseRepository
+      .createQueryBuilder('release')
+      .innerJoin(
+        'release.promotions',
+        'promotion',
+        'promotion.channel = :channel',
+        {channel: 'rollback'},
+      )
+      .select('release.name')
+      .addSelect('promotion.channel')
+      .getMany();
+
+    const channelQueries = [
```

Comment:
Instead of querying releases and joining on promotions, why not query the promotions table directly? This way I think you can avoid a join because the promotions table will have `release.name` as a foreign key.

Reply:
But I thought `release.name` isn't the foreign key, `id` is. We should talk about this tomorrow morning. I'm confused.

Reply:
for example this is all the information I have accessible to me without joining in a promotion `{id: 22, channel: 'nightly', date: 2020-04-17T01:48:43.000Z}`

Reply:
Does this work now with the joinColumnOptions you added to the promotion entity? 🎉

Reply:
YES! :)